libsolidity/interface/CompilerStack.cpp

```diff
@@ -180,7 +181,27 @@ bool CompilerStack::parse()
 				if (!resolver.updateDeclaration(*m_globalContext->currentThis())) return false;
 				if (!resolver.updateDeclaration(*m_globalContext->currentSuper())) return false;
 				if (!resolver.resolveNamesAndTypes(*contract)) return false;
-				m_contracts[contract->name()].contract = contract;
+
+				if (m_contracts.find(contract->fullyQualifiedName()) != m_contracts.end())
```

Comment:
I checked this and the other place again: The compiler returns an error, which means it detects the double declaration, but the aim for the compiler is to still continue as much as possible even if there was an error.

If you leave this code in, the error will be reported twice. So please change this (and the other place) to silently drop the second contract and not store it in `m_contracts`.